### PR TITLE
removed pipeline dependency logic and lock down if not v2

### PR DESCRIPTION
**Pull request by Gkrumbach07**
closes: https://issues.redhat.com/browse/RHOAIENG-2136

## Description

- remove operator dependency logic in both frontend and backend
- add blockers for incompatibility
- make it so we make v2 instead of v1

## How Has This Been Tested?

1. make sure you have a dspa that is v1 (does not have the `dspVersion: 'v2'`)
2. get on this pr
3. Test that pipelines page, pipeline runs page, and pipeline section in project details all have a message saying that you have an incompatible pipeline server
4. delete pipeline server
5. make a new one... you should see the error anymore. You will see a bunch of other errors which is expected because the code is still looking for the old dspa api server routes which dont exist. So as long as you dont get the incompatible warning.

## Test Impact

Added tests to check if warning comes up and removed failing playwrite tests
## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

_This has UI changes but they are temp changes and will need UX refinement later_

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by gregsheremeta** [comment]
I tested this and verified that it works.

File: frontend/src/concepts/pipelines/EnsureCompatiblePipelineServer.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    return <NoPipelineServer variant="secondary" />;
+  }
+
+  if (!pipelinesServer.compatible) {
+    return (
+      <Bullseye data-testid="incompatible-pipelines-server">
+        <EmptyState variant={EmptyStateVariant.lg}>
+          <EmptyStateHeader>
+            The data science pipeline version installed is incompatible. Delete the DSPA and
+            recreate
```
Comment (gregsheremeta) [inline frontend/src/concepts/pipelines/EnsureCompatiblePipelineServer.tsx:39]:
```suggestion
            This data science project contains an unsupported version of Data Science Pipelines.
            Delete the pipeline server and recreate it.
```

Comment (gregsheremeta) [reply frontend/src/concepts/pipelines/EnsureCompatiblePipelineServer.tsx:39]: Probably want docs to give the microcopy here long term, but I at least wanted to remove mention of "DSPA" because users won't know what that is

Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/EnsureCompatiblePipelineServer.tsx:39]: Yeah this UX will be iterated on, so this is more of a placeholder. I can commit your suggestion tho.

File: frontend/src/pages/dependencies/DependencyMissingPage.tsx [hunk 2]
```diff
@@ -1,31 +0,0 @@
...
-    case 'pipelines':
-      if (dashboardConfig.status.dependencyOperators.redhatOpenshiftPipelines.available) {
-        // eslint-disable-next-line no-console
-        console.warn(
-          'The pipelines dependency has been met, this page will redirect to pipelines page.',
-        );
-        navigate('/pipelines');
-        break;
-      }
-      return <PipelinesDependencyMissing />;
```
Comment (andrewballantyne) [inline frontend/src/pages/dependencies/DependencyMissingPage.tsx:22]: Maybe just remove this case and put a "TODO: add more use-cases" -- because honestly this is probably not the last time we will deal with this scenario.

File: frontend/src/concepts/pipelines/content/configurePipelinesServer/utils.ts [hunk 3]
```diff
@@ -100,11 +100,16 @@ export const createDSPipelineResourceSpec = (
       awsRecord.AWS_S3_ENDPOINT?.match(/^(?:(\w+):\/\/)?(.*)/) ?? [];
 
     return {
+      dspVersion: 'v2',
+      mlpipelineUI: {
+        image: 'quay.io/opendatahub/ds-pipelines-frontend:latest', // TODO: remove this before release
+      },
```
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/configurePipelinesServer/utils.ts:106]: We should log a Task for this effort so it's associated to the Epic.

File: frontend/src/concepts/pipelines/content/configurePipelinesServer/utils.ts [hunk 4]
```diff
@@ -100,11 +100,16 @@ export const createDSPipelineResourceSpec = (
...
       objectStorage: {
         externalStorage: {
           host: externalStorageHost?.replace(/\/$/, '') || '',
           scheme: externalStorageScheme || 'https',
           bucket: awsRecord.AWS_S3_BUCKET || '',
+          region: 'us-east-2', // TODO hardcode for now
```
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/configurePipelinesServer/utils.ts:112]: @HumairAK Do you think we will need this with v2 launch DSPA -- should we look to make it a field? Or is this something we want to get rid of and it's just a process now?

File: frontend/src/concepts/pipelines/EnsureCompatiblePipelineServer.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    return <NoPipelineServer variant="secondary" />;
+  }
+
+  if (!pipelinesServer.compatible) {
+    return (
+      <Bullseye data-testid="incompatible-pipelines-server">
+        <EmptyState variant={EmptyStateVariant.lg}>
+          <EmptyStateHeader>
+            This data science project contains an unsupported version of Data Science Pipelines.
+            Delete the pipeline server and recreate it.
```
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/EnsureCompatiblePipelineServer.tsx:39]: There is talk about adding a simple to click CTA here -- but we can iterate on it as they work it out.

cc @yannnz FYI.

File: frontend/src/app/AppRoutes.tsx [hunk 6]
```diff
@@ -70,8 +66,6 @@ const AppRoutes: React.FC = () => {
         <Route path="/pipelines/*" element={<GlobalPipelinesRoutes />} />
         <Route path="/pipelineRuns/*" element={<GlobalPipelineRunsRoutes />} />
 
-        <Route path="/dependency-missing/:area" element={<DependencyMissingPage />} />
```
Comment (andrewballantyne) [inline frontend/src/app/AppRoutes.tsx:73]: I'm on the fence, but I think it would be great to keep this here -- I know it's kinda goofy, but I think it has low harm too... and showcases how to do it in the future if we need it.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

**Comment by andrewballantyne** [comment]
We should definitely run this one through UX

image

cc @yannnz thoughts?

**Comment by andrewballantyne** [comment]
This is pretty good -- I think just a couple alterations to the code and we should be good to merge this.

File: frontend/src/app/AppRoutes.tsx (hunk 6 above)
Comment (Gkrumbach07) [reply frontend/src/app/AppRoutes.tsx]: brought it back

File: frontend/src/concepts/pipelines/content/configurePipelinesServer/utils.ts (hunk 3 above)
Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/content/configurePipelinesServer/utils.ts:106]: https://issues.redhat.com/browse/RHOAIENG-2340

File: frontend/src/concepts/pipelines/content/configurePipelinesServer/utils.ts (hunk 4 above)
Comment (HumairAK) [reply frontend/src/concepts/pipelines/content/configurePipelinesServer/utils.ts:112]: The region thing is a bit tricky, we definitely need the user to specify the region, but we also need it in the `host` field  in the form of `s3.<region>.amazonaws.com`. This is all for aws s3.

(this is due to a bug I encountered in kfp, where it was resulting in some mismatched headers if we didn't specify the region in the endpoint for aws s3) 

I'm not very well versed on other s3 providers and how it works for them (for example gs/ceph/etc), it would be nice if we can figure out a list of providers that we support so I can go and test them all out, who has a region as required field or not.

Comment (andrewballantyne) [reply frontend/src/concepts/pipelines/content/configurePipelinesServer/utils.ts:112]: Ah this is interesting -- so we will need a story to chase this down -- Guillaume has been a good source, we should confer with him on the ticket.

@Gkrumbach07 mind logging a Task for this one too?

File: frontend/src/app/AppRoutes.tsx [hunk 7]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import { Route, Routes } from 'react-router-dom';
 import ApplicationsPage from '~/pages/ApplicationsPage';
 import UnauthorizedError from '~/pages/UnauthorizedError';
+import DependencyMissingPage from '~/pages/dependencies/DependencyMissingPage';
```
Comment (andrewballantyne) [inline frontend/src/app/AppRoutes.tsx:5]: Need to revert the changes in this file -- we need the lazy import for route chunking.

Comment (Gkrumbach07) [reply frontend/src/app/AppRoutes.tsx:5]: arg missed that. will do

**Review by andrewballantyne — APPROVED** [review]
LGTM, just need those extra tickets

File: frontend/src/concepts/pipelines/EnsureCompatiblePipelineServer.tsx (hunk 5 above)
Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/EnsureCompatiblePipelineServer.tsx:39]: https://issues.redhat.com/browse/RHOAIENG-2389

File: frontend/src/concepts/pipelines/content/configurePipelinesServer/utils.ts (hunk 4 above)
Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/content/configurePipelinesServer/utils.ts:112]: https://issues.redhat.com/browse/RHOAIENG-2391
